framework/contrib/ice_updater/include/LibcurlUtils.h

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
```

Comment:
I suspect Curl is "hard" to build (meaning it can't just be put in `contrib`) (probably has it's own `configure` that really does need to be run).

Now that I look at the rest of this stuff... I actually wouldn't mind all of the ICE agnostic stuff just being directly in `framework/include/utils` or maybe `framework/include/utils/url` or something.  It already uses a lot of MOOSE stuff and it's weird that we now have stuff in `contrib` that is #including MOOSE stuff anyway.  It should really probably just be part of the framework.

That said, if we do that the license (at least on these parts) will definitely need to change to LGPL 2.1...